display.go

```diff
@@ -12,15 +12,51 @@ import (
 	"fyne.io/fyne/v2/canvas"
 	"fyne.io/fyne/v2/container"
 	"fyne.io/fyne/v2/driver/desktop"
+	"fyne.io/fyne/v2/widget"
+	"tinygo.org/x/drivers/touch"
 )
 
+type touchWidget struct {
+	widget.BaseWidget
+	obj fyne.CanvasObject
+	d   *Device
+}
+
+func (s *touchWidget) CreateRenderer() fyne.WidgetRenderer {
+	return widget.NewSimpleRenderer(s.obj)
+}
+
+func (s *touchWidget) Tapped(p *fyne.PointEvent) {
+	s.d.mu.Lock()
+
+	s.d.TouchPoint.X = int(p.AbsolutePosition.X) * ((1 << 16) / s.d.Width)
+	s.d.TouchPoint.Y = int(p.AbsolutePosition.Y) * ((1 << 16) / s.d.Height)
+	s.d.TouchPoint.Z = 1 << 16
+	s.d.mu.Unlock()
+}
+
+func (s *touchWidget) Dragged(d *fyne.DragEvent) {
+	s.d.mu.Lock()
+	s.d.TouchPoint.X = int(d.AbsolutePosition.X) * ((1 << 16) / s.d.Width)
+	s.d.TouchPoint.Y = int(d.AbsolutePosition.Y) * ((1 << 16) / s.d.Height)
```

Comment:
At least in my Windows environment, it sometimes takes values outside the Window range.
For example, a value less than 0, or a value greater than s.d.Width.

It would be better if you set an upper and lower limit for the value.

Reply:
Same on Linux I didn't notice in my test. I've clamped the values between 0 width and height.